src/cascadia/TerminalSettingsModel/ActionMap.cpp

```diff
@@ -369,38 +363,41 @@ namespace winrt::Microsoft::Terminal::Settings::Model::implementation
     {
         auto actionMap{ make_self<ActionMap>() };
 
-        // copy _KeyMap (KeyChord --> ID)
+        // KeyChord --> ID
         actionMap->_KeyMap = _KeyMap;
 
-        // copy _ActionMap (ID --> Command)
+        // ID --> Command
+        actionMap->_ActionMap.reserve(_ActionMap.size());
         for (const auto& [actionID, cmd] : _ActionMap)
         {
-            actionMap->_ActionMap.emplace(actionID, *(get_self<Command>(cmd)->Copy()));
+            actionMap->_ActionMap.emplace(actionID, *winrt::get_self<Command>(cmd)->Copy());
         }
 
-        // copy _MaskingActions (ID --> Command)
+        // ID --> Command
+        actionMap->_MaskingActions.reserve(_MaskingActions.size());
         for (const auto& [actionID, cmd] : _MaskingActions)
         {
-            actionMap->_MaskingActions.emplace(actionID, *(get_self<Command>(cmd)->Copy()));
+            actionMap->_MaskingActions.emplace(actionID, *winrt::get_self<Command>(cmd)->Copy());
         }
 
-        // copy _NestedCommands (Name --> Command)
+        // Name --> Command
+        actionMap->_NestedCommands.reserve(_NestedCommands.size());
         for (const auto& [name, cmd] : _NestedCommands)
         {
-            actionMap->_NestedCommands.Insert(name, *(get_self<Command>(cmd)->Copy()));
+            actionMap->_NestedCommands.emplace(name, *winrt::get_self<Command>(cmd)->Copy());
         }
 
-        // copy _IterableCommands
+        actionMap->_IterableCommands.reserve(_IterableCommands.size());
         for (const auto& cmd : _IterableCommands)
         {
-            actionMap->_IterableCommands.Append(*(get_self<Command>(cmd)->Copy()));
+            actionMap->_IterableCommands.emplace_back(*winrt::get_self<Command>(cmd)->Copy());
         }
 
-        // repeat this for each of our parents
-        FAIL_FAST_IF(_parents.size() > 1);
+        assert(_parents.size() <= 1);
+        actionMap->_parents.reserve(_parents.size());
         for (const auto& parent : _parents)
         {
-            actionMap->_parents.emplace_back(parent->Copy());

```

Comment:
Seems like removing this `Copy()` call is unexpected? All the other `Copy()`s were left untouched - why this one?

Reply:
Ah thank you for catching this! I've fixed it now.